ngraph/core/reference/include/ngraph/runtime/reference/roi_align.hpp

```diff
@@ -160,10 +160,10 @@ void roi_align(const T* feature_maps,
 
                         switch (pooling_mode) {
                         case ROIPoolingMode::MAX: {
-                            T sample_value = std::max({pooling_weights[sample_index] * sample_part_1,
-                                                       pooling_weights[sample_index + 1] * sample_part_2,
-                                                       pooling_weights[sample_index + 2] * sample_part_3,
-                                                       pooling_weights[sample_index + 3] * sample_part_4});
+                            T sample_value = pooling_weights[sample_index] * sample_part_1 +
+                                             pooling_weights[sample_index + 1] * sample_part_2 +
+                                             pooling_weights[sample_index + 2] * sample_part_3 +
+                                             pooling_weights[sample_index + 3] * sample_part_4;
```

Comment:
Seems `sample_value` for `MAX` mode can be calculated same way like for `AVG` mode and only `pooled_value` calculation should make difference, so we can move `sample_value` calculation above the mode switch.